editor/app/src/main/java/ca/ubc/eml/soiltopargraphy/editor/ui/main/MainFragment.kt

```diff
@@ -86,14 +97,24 @@ class MainFragment : Fragment(), OnMapReadyCallback, GoogleMap.OnMapLongClickLis
             activity!!.invalidateOptionsMenu()
         }
 
-        // Button to add a new flag
-        val addFlag = view.findViewById<View>(R.id.addButton)
-        addFlag.setOnClickListener {
 
-            //Add flag to centre of screen
+        //Update the database with new flag information when the save button is clicked
+        val saveFlag = view.findViewById<Button>(R.id.saveButton)
+        saveFlag.setOnClickListener {
+
+            //get list of flags from database
+            val db = AppDatabase.getDatabase(Application())
 
+            val mFlagDao = db.flagDao()
+            val flags = mFlagDao.allFlags
 
+            val mainViewModel = MainViewModel(Application())
 
+            if (flags != null){
+                for (flag in flags as List<Flag> ){
```

Comment:
Doing an invalid cast like this will crash the app. Instead, get the `List` value from the `LiveData` class.
```suggestion
                for (flag in flags.value) {
```